### PR TITLE
Update configuration for the buildah

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

Switches to the new stable buildah release.
Makes it possible to build the image inside running container.

Partially fixes https://github.com/eclipse/che/issues/15035

Screenshot from 2020-03-03 15-42-20

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,13 +8,15 @@
 # Contributors:
 #   Red Hat, Inc. - initial API and implementation
 
-FROM quay.io/buildah/stable:v1.12.0
+FROM quay.io/buildah/stable:v1.14.0
 
 ENV KUBECTL_VERSION="v1.17.2" \
```
Comment (ericwill) [inline Dockerfile]: Should be 1.17.3

File: etc/storage.conf [hunk 2]
```diff
@@ -1,31 +1,137 @@
+# This file is is the configuration file for all tools
```
Comment (ericwill) [inline etc/storage.conf]: double "is"

File: etc/storage.conf [hunk 3]
```diff
@@ -1,31 +1,137 @@
...
 
+# Default Storage Driver
+driver = "vfs"
+
+# Temporary storage location
+# runroot = "/var/run/containers/storage"
+runroot = "/var/tmp/theia"
+
+# Primary Read/Write location of container storage
+# graphroot = "/var/lib/containers/storage"
```
Comment (ericwill) [inline etc/storage.conf]: Is this one intentionally commented out?

File: etc/storage.conf [hunk 4]
```diff
@@ -1,31 +1,137 @@
...
+# and use any image even those with multiple uids.  Note multiple UIDs will be
+# squashed down to the default uid in the container.  These images will have no
+# separation between the users in the container. Only supported for the overlay
+# and vfs drivers.
+#ignore_chown_errors = false
+ignore_chown_errors = "true"
+
+# Path to an helper program to use for mounting the file system instead of mounting it
+# directly.
+#mount_program = "/usr/bin/fuse-overlayfs"
```
Comment (ericwill) [inline etc/storage.conf]: If we are not using fuse-overlayfs it's probably best to remove this.

File: Dockerfile [hunk 5]
```diff
@@ -8,13 +8,15 @@
...
     HELM_VERSION="v3.0.3" \
```
Comment (ericwill) [inline Dockerfile]: Let's update this to 3.1.1 as well.

**Review by ericwill — CHANGES_REQUESTED** [review]

File: etc/storage.conf (hunk 2 above)
Comment (vitaliy-guliy) [reply etc/storage.conf]: Fixed

File: etc/storage.conf (hunk 3 above)
Comment (vitaliy-guliy) [reply etc/storage.conf]: Yes, but has been removed.

File: etc/storage.conf (hunk 4 above)
Comment (vitaliy-guliy) [reply etc/storage.conf]: I have copied the default configuration file and then changed necessary properties. I think it's a good practice, isn't it?

Comment (vitaliy-guliy) [reply etc/storage.conf]: But I'm also Ok to cleanup the file and left here only few lines.

File: Dockerfile (hunk 1 above)
Comment (vitaliy-guliy) [reply Dockerfile]: Done

File: Dockerfile (hunk 5 above)
Comment (vitaliy-guliy) [reply Dockerfile]: Done

File: etc/storage.conf (hunk 4 above)
Comment (ericwill) [reply etc/storage.conf]: Let's keep it clean just for sake of consistency and making it easier to read.

**Review by ericwill — APPROVED** [review]
Nice. Let's do the same for the 1.1.0 branch too.

**Pull request merged**